pkg/generate/app/app.go

```diff
@@ -216,6 +219,9 @@ func (r *ImageRef) RepoName() string {
 
 // SuggestName suggests a name for an image reference
 func (r *ImageRef) SuggestName() (string, bool) {
+	if r != nil && len(r.ObjectName) > 0 {
```

Comment:
I want ObjectName to take precedence over whatever's in r.Name, so I can't move the bottom one on top. I could add the if r == nil { return "", false } by itself first though. Not sure that makes it better.